drivers/gpio/gpio_ambiq.c

```diff
@@ -161,17 +321,23 @@ static int ambiq_gpio_port_get_raw(const struct device *dev, gpio_port_value_t *
 {
 	const struct ambiq_gpio_config *const dev_cfg = dev->config;
 
-	*value = (*AM_HAL_GPIO_RDn(dev_cfg->offset >> 2));
-
+#if defined(CONFIG_SOC_SERIES_APOLLO3X)
+	*value = AM_REGVAL(AM_REGADDR(GPIO, RDA) + (dev_cfg->offset >> 3));
+#else
+	*value = AM_REGVAL(AM_REGADDR(GPIO, RD0) + (dev_cfg->offset >> 5));
+#endif
```

Comment:
Should restrict the use of AM_REGVAL only in HAL. Add a AM_HAL_GPIO_RDn if it not already there.